app/models/work.rb

```diff
@@ -0,0 +1,20 @@
+class Work < ApplicationRecord
+  has_many :votes
+  has_many :users, through: :votes
+
+  CATEGORIES = %w(movie book album)
+
+  validates :title, presence: true, uniqueness: true
+  validates :creator, presence: true
+  validates :publication_year, presence: true, :numericality => {:less_than => Date.today.year, :greater_than => 1700}
```

Comment:
Nice adding this validation... But did the original site have this validation on it?